src/Mapbender/RoutingBundle/Resources/public/mapbender.element.routing.js

```diff
@@ -0,0 +1,764 @@
+            var lineGeometry = routeData.features;
+            var self = this;
+            lineGeometry.forEach(function (feature, index) {
+                var geomCoordinates = feature.geometry.coordinates;
+                var propWayPoints = feature.properties.waypoints || null;
+
+                routeData.features[index].geometry.coordinates = self._transformLineGeometry(
+                    geomCoordinates,
+                    respProj,
+                    mapProj
+                );
+
+                // transform snapped WayPoints
+                if (propWayPoints){
+                    propWayPoints.forEach(function (element) {
+                        element.coordinates = self._transformCoordinates(
+                            element.coordinates,
+                            respProj,
+                            mapProj
+                        );
+                        element.srid = mapProj.projCode;
+                    });
+                }
+            });
+        },
+
+        _transformLineGeometry: function(lineGeometry, inputProj, mapProj) {
+            var transformedLineGeometry = [];
+            var self = this;
+            lineGeometry.forEach(function(pt) {
+                var p = self._transformCoordinates(pt, inputProj, mapProj);
+                transformedLineGeometry.push([p[0], p[1]]);
+            });
+            return transformedLineGeometry;
+        },
+
+        _transformMultiLineGeometry: function(multilineGeometry, requestProj, mapProj) {
+            var self = this;
+            var transformedLineGeometry = [];
+            multilineGeometry.forEach(function(line) {
+                var l = self._transformLineGeometry(line, requestProj, mapProj);
+                transformedLineGeometry.push(l);
+            });
+            return transformedLineGeometry;
+        },
+
+        _revGeocode: function (feature){
+            var self = this;
+
+            if (self.options.useReverseGeocoding) {
+                var p = {
+                    name: "point",
+                    value: [feature.geometry.x,feature.geometry.y]
+                };
+
+                self._getRevGeocode([p]).then(function(response) {
+                    var resultLabel = self._checkResultLabel(feature,response);
+                    $(feature.attributes.input)
+                        .val(resultLabel)
+                        .data('coords', [feature.geometry.x, feature.geometry.y])
+                        .change();
+                });
+
+                return true;
+
+            }else{
+                return false;
+            }
+        },
+
+        _getRevGeocode: function(coordinate) {
+            var self = this;
+            return $.ajax({
+                type: "GET",
+                url: self.elementUrl + 'revGeocode',
+                data: {
+                    coordinate: coordinate,
+                    srsId: self.olMap.getProjectionObject().projCode
```

Comment:
Reverse Geocoding fails in my test. Error is `Uncaught TypeError: self.olMap.getProjectionObject is not a function`

Should probably use `self.olMap.getView().getProjection().getCode()` (untested though)